### PR TITLE
application.conf and reference.conf

### PR DESCRIPTION
**Pull request by jamesmcclain**
Changes:
   * Various configuration files are removed in this pull request.  Those files which are required to compile and pass tests have been retained untouched.
   * A stray file from the deleted `vector-benchmarks` project has been removed.
   * A typo in the recent `BufferTiles` fixes, https://github.com/geotrellis/geotrellis/pull/1352, is corrected

### CODE REVIEW

File: spark/src/main/resources/reference.conf [hunk 1]
```diff
@@ -1 +0,0 @@
-geotrellis.accumulo.catalog = "metadata"
```
Comment (lossyrob) [inline spark/src/main/resources/reference.conf]: We actually need this. This is a default that will be pulled into the user's `application.conf`, even if they don't supply one or unless they overwrite it.

File: raster/src/main/resources/reference.conf [hunk 2]
```diff
@@ -1,11 +0,0 @@
...
-include "akka-remote-reference"
-
-# Set the geotrellis catalog here.
-# Required for reading any data from GeoTrellis by name.
-#geotrellis.catalog = "src/test/resources/catalog.json"
-
-akka {
-  log-dead-letters = 0
-  log-dead-letters-during-shutdown = off
-}
```
Comment (lossyrob) [inline raster/src/main/resources/reference.conf]: This `reference.conf` should actually be moved to the `engine` project's not-yet-existing `reference.conf`

Comment (jamesmcclain) [inline raster/src/main/resources/reference.conf]: Okay, will-do.

File: spark/src/main/resources/reference.conf (hunk 1 above)
Comment (jamesmcclain) [inline spark/src/main/resources/reference.conf]: It shall be restored.

Comment (lossyrob) [inline spark/src/main/resources/reference.conf]: sorry...this actually doesn't belong in `spark`, but belongs in `accumulo` now

Comment (jamesmcclain) [inline spark/src/main/resources/reference.conf]: Okay, it shall go to accumulo.

**Pull request merged**